### PR TITLE
Feat/cse 148 content nav

### PR DESCRIPTION
**Pull request by henrypeterson**
# Changes introduced

This branch creates the content navigator which will be on the target page. It will allow users to navigate between sections quicker.

## Related issue(s):

- CSE-255
- CSE-148

## Contributor checklist

- [x] Wrote/updated tests for introduced changes
- [x] Added new stories for created/modified components (if applicable)
- [x] Checked Storybook for Accessibility issues (if applicable)
- [ ] ~~Updated Postman library for created/modified routes (if applicable)~~
- [x] Verified that there are no issues in CircleCI
- [x] Assigned yourself to the PR
- [x] Removed `WIP:` from the PR title
- [x] Requested review from the SB Reviewers team

## Reviewer checklist

- [x] Assigned yourself to the PR
- [x] Read through the changes in the `Files changed` tab
- [x] Verified that tests were written/modified
- [x] Verified that stories were written/modified (if applicable)
- [ ] ~~Verified that Postman was updated (if applicable)~~
- [x] Checked out the branch and tested changes locally
- [ ] Run storybook and checked that there are no accessibility issues

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -127,6 +127,7 @@
     "prettier": "^1.14.2",
     "react-test-renderer": "^16.4.2",
     "style-loader": "^0.21.0",
+    "styled-jsx": "^3.0.2",
```
Comment (puregarlic) [inline package.json]: Dunno how this managed to sneak down to `devDependencies`, move it back up to `dependencies`.

File: package.json [hunk 2]
```diff
@@ -71,14 +71,12 @@
   },
   "dependencies": {
     "@sbac/sbac-ui-kit": "github:osu-cass/sbac-global-ui-kit#develop",
-    "@types/react-router": "^4.0.30",
-    "@types/react-router-dom": "^4.3.0",
+    "polyfill": "^0.1.0",
```
Comment (puregarlic) [inline package.json]: ... What is this and how did it get here

File: src/components/ContentNav/ContentNav.storybook.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import centered from '@storybook/addon-centered';
+
+import { ContentNav } from '.';
+import { Item, ItemProps } from './Item';
+import { SubItem, SubItemProps } from './SubItem';
+
+const subMock1: SubItemProps = { name: 'sdgasdfasdf' };
```
Comment (puregarlic) [inline src/components/ContentNav/ContentNav.storybook.tsx]: Move these into a `__mocks__` directory to save some space.

File: src/components/ContentNav/Item/Item.storybook.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import centered from '@storybook/addon-centered';
+
+import { Item, ItemProps } from '.';
+import { SubItem } from '../SubItem';
+
+const subItems = [
```
Comment (puregarlic) [inline src/components/ContentNav/Item/Item.storybook.tsx]: This is a mock, but it's in denial. Put it the same '__mocks__' directory as the other ones.

File: src/components/ContentNav/Item/index.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { Item } from '.';
+
+describe('Item', () => {
+  it('should be active', () => {
```
Comment (puregarlic) [inline src/components/ContentNav/Item/index.spec.tsx]: I would rephrase this to say `it('should be activated on click', () => {` to better describe what you're testing for.

File: src/components/ContentNav/SubItem/index.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { SubItem } from '.';
+
+describe('SubItem', () => {
+  it('should be active', () => {
+    const subItemActive = mount(<SubItem name={'Test Name'} active={true} />);
+
+    subItemActive.simulate('click');
+    expect(subItemActive).toMatchSnapshot();
```
Comment (puregarlic) [inline src/components/ContentNav/SubItem/index.spec.tsx]: We actually don't want to save the entire component as a snapshot; we want to render the component to JSON first, and then save that as a snapshot. Use the `toJson` helper in the `enzyme-to-json` package to accomplish this. Looking at the generated snapshot for the test should tell you why we would want to do this.

Comment (puregarlic) [reply src/components/ContentNav/SubItem/index.spec.tsx]: Here's a link to the package: https://github.com/adriantoine/enzyme-to-json

File: src/components/ContentNav/index.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,208 @@
+import React from 'react';
+import { mount, ReactWrapper } from 'enzyme';
+import { ContentNav } from './index';
+import { Item, Chevron, ItemProps } from './Item';
+import { SubItem } from './SubItem';
+
+const mock: ItemProps[] = [
```
Comment (puregarlic) [inline src/components/ContentNav/index.spec.tsx]: Move this to the `__mocks__` folder too.

File: src/components/ContentNav/index.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+  });
+});
+
+describe('Chevron', () => {
+  it('is clickable', () => {
+    const chevronDown = mount(<Chevron expanded={false} itemName={'test name'} />);
+    const chevronUp = mount(<Chevron expanded={true} itemName={'test name'} />);
+
+    chevronDown.simulate('click');
+    expect(chevronDown).toMatchSnapshot();
```
Comment (puregarlic) [inline src/components/ContentNav/index.spec.tsx]: Serialize these two `expect` statements to JSON as well.

File: src/components/ContentNav/index.tsx [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+                      />
+                    );
+                  })}
+              </Item>
+            );
+          })}
+        </ul>
+        <style jsx>{`
+          * {
+            font-family: 'PT Sans Caption';
```
Comment (puregarlic) [inline src/components/ContentNav/index.tsx]: I thought we were removing the root-level `font-family` rule?

File: src/components/ContentNav/Item/index.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (puregarlic) [inline src/components/ContentNav/Item/index.spec.tsx]: File should be called `Item.spec.tsx`

File: src/components/ContentNav/SubItem/index.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (puregarlic) [inline src/components/ContentNav/SubItem/index.spec.tsx]: File should be called `SubItem.spec.tsx`

**Review by puregarlic — CHANGES_REQUESTED** [review]
Got just a bit of stuff for you to clean up, then this PR is good.

File: package.json (hunk 1 above)
Comment (henrypeterson) [reply package.json]: Which ones specifically are you referring to?

**Comment by henrypeterson** [comment]
Ok i've made updates, let me know if theres anything else that needs to be done.

File: package.json (hunk 1 above)
Comment (puregarlic) [reply package.json]: `styled-jsx`, sorry about that. We're using it during runtime, so it needs to be considered a dependency.

File: src/components/ContentNav/__mocks__/index.tsx [hunk 12]
```diff
@@ -0,0 +1,53 @@
+import { SubItemProps } from '../SubItem';
+import { ItemProps } from '../Item';
+
+export const subMock1: SubItemProps = { name: 'sdgasdfasdf' };
```
Comment (puregarlic) [inline src/components/ContentNav/__mocks__/index.tsx]: Are you actually using this anywhere?

File: src/components/ContentNav/__mocks__/index.tsx [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+export const subMock2: SubItemProps = { name: 'helasd' };
+export const subMocks3: SubItemProps = [
```
Comment (puregarlic) [inline src/components/ContentNav/__mocks__/index.tsx]: I hate long names as much as the next, but this should be called something like `subItemMocks3`, because `subMocks3` just isn't very useful to anyone. Even better, though, is if you can get rid of the first two `SubItem` mocks and just call this `subItemMock`.

File: src/components/ContentNav/__mocks__/index.tsx [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+  {
+    active: false,
+    name: 'Target Evidence Statement'
+  },
+  {
+    active: false,
+    name: 'Appropriate Stems'
+  }
+];
+export const mock: ItemProps[] = [
```
Comment (puregarlic) [inline src/components/ContentNav/__mocks__/index.tsx]: Same situation here. Be more specific; call this `itemMock`.

**Review by puregarlic — CHANGES_REQUESTED** [review]
Minor name changes are needed for the mocks, then this PR has my approval.

File: src/components/ContentNav/__mocks__/index.tsx (hunk 12 above)
Comment (henrypeterson) [reply src/components/ContentNav/__mocks__/index.tsx]: I am not so I removed it.

File: src/components/ContentNav/__mocks__/index.tsx (hunk 13 above)
Comment (henrypeterson) [reply src/components/ContentNav/__mocks__/index.tsx]: The two unused ones are removed and subMocks3 is renamed to subItemMock.

File: src/components/ContentNav/__mocks__/index.tsx (hunk 14 above)
Comment (henrypeterson) [reply src/components/ContentNav/__mocks__/index.tsx]: It is renamed to itemMock.

**Review by puregarlic — APPROVED** [review]
